gst/nnstreamer/tensor_query/tensor_query_common.c

```diff
@@ -537,29 +537,34 @@ nnstreamer_query_server_accept (query_server_handle server_data)
 
 /**
  * @brief [TCP] receive data for tcp server
+ * @return 0 if OK, negative value if error
  */
-static gboolean
+static int
 query_tcp_receive (GSocket * socket, uint8_t * data, size_t size,
-    GCancellable * cancellable)
+    GCancellable * cancellable, int32_t timeout)
 {
   size_t bytes_received = 0;
   ssize_t rret;
   GError *err = NULL;
+
   while (bytes_received < size) {
-    rret = g_socket_receive (socket, (char *) data + bytes_received,
-        size - bytes_received, cancellable, &err);
+    rret =
+        g_socket_receive_with_blocking (socket, (char *) data + bytes_received,
```

Comment:
need to set timeout with g_socket_set_timeout()?
If it is necessary to set timeout, how about adding this?

Reply:
The non-blocking mode applies only to receiving the first command from the server.
Others operate in blocking mode.
The timeout for the other receiving will be added in the next PR.